modules/protocol/src/utils.ts

```diff
@@ -162,6 +170,33 @@ export function getParamsFromUpdate<T extends UpdateType = any>(
   });
 }
 
+export function getTransferFromUpdate(
+  update: ChannelUpdate<typeof UpdateType.create>,
+  channel: FullChannelState,
+): FullTransferState {
+  return {
+    balance: update.details.balance,
+    assetId: update.assetId,
+    transferId: update.details.transferId,
+    channelAddress: update.channelAddress,
+    transferDefinition: update.details.transferDefinition,
+    transferEncodings: update.details.transferEncodings,
+    transferTimeout: update.details.transferTimeout,
+    initialStateHash: hashTransferState(update.details.transferInitialState, update.details.transferEncodings[0]),
```

Comment:
I get nervous whenever I see these `[0]`. Sometimes they might be correct, like in this case, where I think it refers to Alice & Bob. Sometimes not, like when it's asset ids and is lazy.

Consider these readability improvements for sometime in the future:
- `export type AliceBob<T> = [T, T];`
- Change the type of `transferEncodings` from `string[]` to `AliceBob<string>`
- `export const Alice = 0; export const Bob = 1;`
- `update.details.transferEncodings[Alice]`

If you want to get really fancy you could expose the type to be only allowed to be indexed by tagged by opaque Alice & Bob values, which would prevent any `[0]` without loss of performance because it is still `[0]` under the hood.

Also... is it really correct that this is hardcoded to Alice?

Reply:
No this is specifically for `transferEncodings` which are returned from the `RegisteredTransfer` information onchain. The `transferEncodings` are always ordered `[stateEncoding, resolverEncoding]`.

We could still implement these changes for the state/resolver encodings, though

Reply:
Yeah, the fact that I guessed wrong is a datapoint indicating that readability could be improved.